lib/facility_factory.rb

```diff
@@ -0,0 +1,59 @@
+class FacilityFactory
+  attr_reader :facilities
+  
+  def initialize
+    @facilities = []
+  end
+
+  def create_facilities(data_source) 
+    data_source.each do |reg|
+      @facilities << Dmv.new(reg)
+    end
+    @facilities
+  end
+
+  private
+
+  class Dmv
+    attr_reader :dmv_office,
+                :address_li,
+                :address__1,
+                :city,
+                :state,
+                :zip,
+                :phone
+                
+    def initialize(details)
+      # CO Reference
+      if details[:dmv_office] && details[:address_li] && details[:city] && details[:state] && details[:zip] && details[:phone] 
+        @dmv_office = details[:dmv_office]
+        @address_li = details[:address_li]
+        @address__1 = details[:address__1]
+        @city = details[:city]
+        @state = details[:state]
+        @zip = details[:zip]
+        @phone = details[:phone]
+      elsif details[:office_name] && details[:office_type] && details[:public_phone_number] && details[:street_address_line_1] && details[:city] && details[:state] && details[:zip_code]
+        # New York Reference
+        @dmv_office = details[:office_name]
+        @address_li = details[:office_type]
+        @phone = details[:public_phone_number]
+        @address__1 = details[:street_address_line_1]
+        @city = details[:city]
+        @state = details[:state]
+        @zip = details[:zip_code]
+      else
+        # Missouri Reference
+        @dmv_office = details[:type]
+        @address_li = details[:county]
+        @phone = details[:phone]
+        @address__1 = details[:address1]
+        @city = details[:city]
+        @state = details[:state]
+        @zip = details[:zipcode]
```

Comment:
I'm curious why you have your `DMV` class in this file? I don't believe this is necessary.  You already have a DMV class defined in '/lib/dmv.rb'.